Cargo.toml

```
@@ -27,6 +27,7 @@ once_cell = {version="1.18.0"}
 error-stack = {version="0.3.1"}
 async-trait = {version="0.1.73"}
 regex = {version="1.9.3", features=["perf"]}
+tempfile = "3.8.0"
 
 [dev-dependencies]
 rusty-hook = "^0.11.2"
```

```suggestion

[dev-dependencies]
rusty-hook = "^0.11.2"
tempfile = "3.8.0"
```

Thanks :heart: for working on this PR, I really appreciate it :+1:  but I would suggest you to put the `tempfile` dependency under `dev.dependencies` as we don't want to include in the main binary file and only have it included when we run `cargo tests`.

For more information on `dev.dependencies` I would suggest going through these docs here: 
- https://doc.rust-lang.org/rust-by-example/testing/dev_dependencies.html#development-dependencies
- https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#development-dependencies

---

I agree. Thank you for pointing that out!